File: Trackly.API/Models/Tickets/Ticket.cs
using TracklyApi.Models.Assets;
using static TracklyApi.Helpers.EnumHelper;

namespace TracklyApi.Models.Tickets
{
    public class Ticket
    {
        public Guid TicketId { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public TicketStatus Status { get; set; }
        public Priority Priority { get; set; }
        public TicketCategory Category { get; set; }
        public string CreatedBy { get; set; }
        public string? AssignedUserId { get; set; }
        public Guid AssetId { get; set; }

        public DateTime CreatedAt { get; set; }
        public DateTime? CompletedAt { get; set; }
        public DateTime? ClosedAt { get; set; }

        // Navigation properties
        public virtual Asset Asset { get; set; }
    }
}

File: Trackly.APP/trackly_app/lib/src/data/models/Tickets/ticket.dart
import 'package:trackly_app/src/data/enumhelper/enums.dart';

class Ticket {
  final dynamic id;
  final String title;
  final String description;
  final Status status;
  final Priority priority;
  final Category category;
  final String assignedUser;
  final DateTime createdAt;
  final DateTime? completedAt;
  final DateTime? closedAt;

  Ticket({
    required this.id,
    required this.title,
    required this.description,
    required this.status,
    required this.priority,
    required this.category,
    required this.assignedUser,
    required this.createdAt,
    required this.completedAt,
    required this.closedAt,
  });

  factory Ticket.fromJson(Map<String, dynamic> json) => Ticket(
        id: json["id"],
        title: json["title"],
        description: json["description"],
        status: statusValues.map[json["status"]]!,
        priority: priorityValues.map[json["priority"]]!,
        category: categoryValues.map[json["category"]]!,
        assignedUser: json["assignedUser"],
        createdAt: DateTime.parse(json["createdAt"]),
        completedAt: json["completedAt"] == null
            ? null
            : DateTime.parse(json["completedAt"]),
        closedAt:
            json["closedAt"] == null ? null : DateTime.parse(json["closedAt"]),
      );

  Map<String, dynamic> toJson() => {
        "id": id,
        "title": title,
        "description": description,
        "status": statusValues.reverse[status],
        "priority": priorityValues.reverse[priority],
        "category": categoryValues.reverse[category],
        "assignedUser": assignedUser,
        "createdAt": createdAt.toIso8601String(),
        "completedAt": completedAt?.toIso8601String(),
        "closedAt": closedAt?.toIso8601String(),
      };
}
